packages/components/src/VirtualizedList/ListGrid/ListGrid.component.js

```diff
@@ -9,48 +9,47 @@ import theme from './ListGrid.scss';
  * List renderer that accepts a custom row renderer.
  * The row renderer will create a row element for each collection item.
  */
-class ListGrid extends React.Component {
-	render() {
-		const {
-			children,
-			id,
-			height,
-			isSelected,
-			rowHeight,
-			rowRenderer,
-			selectionToggle,
-			width,
-		} = this.props;
+function ListGrid(props) {
+	const {
+		children,
+		id,
+		height,
+		isActive,
+		isSelected,
+		onRowClick,
+		rowHeight,
+		rowRenderer,
+		selectionToggle,
+		width,
+	} = props;
 
-		let enhancedRowRenderer = rowRenderer;
-		if (selectionToggle) {
-			enhancedRowRenderer = getRowSelectionRenderer(
-				rowRenderer,
-				{
-					isSelected,
-					getRowData: ({ index }) => this.props.collection[index],
-				}
-			);
-		}
+	const enhancedRowRenderer = selectionToggle || onRowClick ?
```

Comment:
Please don't use mutli line ternaries. It would ba a good idea to extract it's logic into a function.